docs: Add Agile Development pillars (#638)
microsoft/code-with-engineering-playbook: docs/agile-development/delivery-plan/README.md

# Delivery Plan

## Goals

> While Scrum does not require, and in fact discourages planning more than one sprint at a time, most of us work in enterprises where we are dependent on each other. For example: marketing, sales, and support needs to have know what is coming up so they can make plans for budget, people and resources. A delivery plan helps to ensure your teams are aligning with your organizational goals.

Reviewer: Nit: would suggest using "Agile" vs "Scrum" as different teams tend to use different versions of Agile, not just Scrum
```suggestion
While Agile does not require, and in fact discourages planning more than one sprint at a time, most of us work in enterprises where we are dependent on each other. For example: marketing, sales, and support needs to have know what is coming up so they can make plans for budget, people and resources. A delivery plan helps to ensure your teams are aligning with your organizational goals.
```

A rough assessment of the planned project scope should be achievable within a reasonable time frame leveraging the available resources. The goal is to have a rough plan and estimate as a starting point, not to implement "Agilefall."

Note that this is just a starting point to enable planning discussions. We expect the actual schedule to evolve and shift over time, and that you will update the scope and timeline as you progress.

## Benefits

- As you complete the assessment, you can refine the scope, time frame, or ask for additional resources.

- As you progress in your project/product delivery, you can highlight risks to the scope, time frame, and resources.

## Approach

One approach you can take to accomplish is with stickies and a spreadsheet.

Step 1: Stack rank the features for EVERYTHING in the backlog; including

- Functional Features

> - [Non-functional Features](docs/TECH-LEADS-CHECKLIST.md)

Reviewer 2: Remove this line, it's a dead link and out of context

Reviewer 3: ```suggestion
```

- User Research and Design (e.g. research, mockups, prototype, usability studies)
- Testing
- Documentation
- Knowledge Transfer/Support Processes

Step 2: T-Shirt features in terms of working weeks per person. In some scenarios, you have no idea how complex the work may be. In this situation, you can ask for time to conduct a spike (timebox the effort spend on spikes).

Step 3: Calculate the capacity for the team based on the number of weeks person with his/her start and end date and minus holidays, vacation, conferences, training, and onboarding days. Also, minus time if the person is also working on defects and support.

Step 4: Based on your capacity, you now have additional options:

- Ask for more resources. Caution: onboarding new resources takes time.

> - Reduce the scope to the most MVP.  Caution: as you trim more of the scope, you risk negatively impacting the project's value propositions. Consider a cupcake which is everything you need. You don't want to skim on the frosting.

Reviewer: ```suggestion
- Reduce the scope to an MVP.  Caution: as you trim more of the scope, you risk negatively impacting the project's value propositions. Consider a cupcake which is everything you need. You don't want to skim on the frosting.
```

> - Ask for more time. Usually, this is the most flexible, but if there is a marketing date that you need to hit, this might be a flexiblity point.

Reviewer: Suggest making the marketing date an example as it's not something we deal with very often in CSE. 
```suggestion
- Ask for more time. Usually, this is the most flexible, but depends on the customer (there could be a fixed marketing date for a new product).
```
